### PR TITLE
Add user service with tests, added mock on storage

### PR DESCRIPTION
**Pull request by stasBigunenko**
Add user service with tests.
Addition added mock storage

### CODE REVIEW

**Comment by stasBigunenko** [comment]
should make fixes

File: service/user/service.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+func NewUsrService(s newStorage.NewStore) *UsrService {
+	return &UsrService{
+		storage: s,
+	}
+}
+
+func (u *UsrService) Get(_ context.Context, id uuid.UUID) (model.UserHTTP, error) {
+
+	res, err := u.storage.Get(context.Background(), id)
```
Comment (DrIhor) [inline service/user/service.go]: Can pass param context as storage value

File: service/user/service.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+	if res == nil && err == nil || err != nil {
```
Comment (DrIhor) [inline service/user/service.go]: Can split to error check and res check(2 different if). Also do not check `err == nil || err != nil` - same conditions

File: service/user/service.go [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+		ac = append(ac, u.Us)
+	}
+	return ac, nil
+}
+
+func (u *UsrService) Create(_ context.Context, b string) (model.UserHTTP, error) {
+
+	for _, r := range b {
+		if !unicode.IsLetter(r) {
```
Comment (DrIhor) [inline service/user/service.go]: Name can be **Bob0001**. You don`t need to check name.

File: service/user/service.go [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+		if err != nil {
+			return nil, errors.New("unmarshal problem")
+		}
+
+		ac = append(ac, u.Us)
+	}
+	return ac, nil
+}
+
+func (u *UsrService) Create(_ context.Context, b string) (model.UserHTTP, error) {
```
Comment (DrIhor) [inline service/user/service.go]: Can update value names(b is strange choose)

Comment (DrIhor) [reply service/user/service.go]: Not important)

**Comment by DrIhor** [comment]
All comments can be used to all functions.

File: service/user/service.go (hunk 2 above)
Comment (stasBigunenko) [reply service/user/service.go]: forgot brackets

File: service/user/service.go (hunk 3 above)
Comment (stasBigunenko) [reply service/user/service.go]: deleted check name

File: service/user/service.go (hunk 4 above)
Comment (stasBigunenko) [reply service/user/service.go]: updated value to "name"

File: service/user/service.go (hunk 1 above)
Comment (stasBigunenko) [reply service/user/service.go:25]: i think better to send id value directly. Context.Value data should inform us and not control input value for the function.

File: service/user/service.go [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+	if err != nil {
+		return model.UserHTTP{}, errors.New("marshal problem")
+	}
+
+	res, err := u.storage.Update(context.Background(), id, bt)
+	if err != nil || (res == nil && err == nil) {
+		return model.UserHTTP{}, errors.New("not found")
+	}
+
+	err = json.Unmarshal(res, &u.Us)
```
Comment (romanilchyshyn) [inline service/user/service.go]: WTF is this?

File: service/user/service.go [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+	"context"
+	"encoding/json"
+	"errors"
+	"github.com/google/uuid"
+	"github.com/stasBigunenko/monorepa/model"
+	"github.com/stasBigunenko/monorepa/pkg/storage/newStorage"
+)
+
+type UsrService struct {
+	Us      model.UserHTTP
```
Comment (romanilchyshyn) [inline service/user/service.go]: Why we need this?

File: service/user/service.go (hunk 5 above)
Comment (stasBigunenko) [reply service/user/service.go]: unmarshaling []byte received from storage in model.UserHHTP that we should return

File: service/user/service.go [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+	if err != nil {
+		return model.UserHTTP{}, errors.New("storage problem")
+	}
+
+	err = json.Unmarshal(res, &u.Us)
+	if err != nil {
+		return model.UserHTTP{}, errors.New("unmarshal problem")
+	}
+
+	u.Us.ID = id
```
Comment (romanilchyshyn) [inline service/user/service.go]: Why we cannot unmarshal id too?

File: service/user/service.go (hunk 6 above)
Comment (stasBigunenko) [reply service/user/service.go]: we return model.UserHHTP,in most of function, i thought would be easily with return

File: service/user/service.go (hunk 7 above)
Comment (stasBigunenko) [reply service/user/service.go]: with create func of storage we only creating id storage, and storage don't know what is the model will be.
In services we will create fields of the model

File: service/user/service.go (hunk 6 above)
Comment (romanilchyshyn) [reply service/user/service.go]: no, this shouldn't be a service field, local var only. imagine what if we use this service in parallel and many goroutines want to rewrite this

Comment (stasBigunenko) [reply service/user/service.go]: done

File: service/user/service.go [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+func NewUsrService(s newStorage.NewStore) *UsrService {
+	return &UsrService{
+		storage: s,
+	}
+}
+
+func (u *UsrService) Get(_ context.Context, id uuid.UUID) (model.UserHTTP, error) {
+
+	res, err := u.storage.Get(context.Background(), id)
+	if (res == nil && err == nil) || err != nil {
```
Comment (DrIhor) [inline service/user/service.go]: `If err != nil` can be other errors. Add one more if.

File: service/user/service.go [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+		}
+
+		ac = append(ac, m)
+	}
+	return ac, nil
+}
+
+func (u *UsrService) Create(_ context.Context, name string) (model.UserHTTP, error) {
+
+	if len(name) <= 2 {
```
Comment (DrIhor) [inline service/user/service.go]: This check is not important too.

File: service/user/service.go [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+	}
+
+	res, id, err := u.storage.Create(context.Background(), bt)
+	if err != nil {
+		return model.UserHTTP{}, errors.New("storage problem")
+	}
+
+	err = json.Unmarshal(res, &m)
+	if err != nil {
+		return model.UserHTTP{}, errors.New("unmarshal problem")
```
Comment (DrIhor) [inline service/user/service.go:82]: In this case can just return err)

File: service/user/service.go (hunk 1 above)
Comment (DrIhor) [reply service/user/service.go]: Change it to control full work line.

File: service/user/service.go (hunk 8 above)
Comment (stasBigunenko) [reply service/user/service.go]: done

File: service/user/service.go (hunk 9 above)
Comment (stasBigunenko) [reply service/user/service.go]: deleted this check

File: service/user/service.go (hunk 10 above)
Comment (stasBigunenko) [reply service/user/service.go:82]: some errors changes according the comment

File: service/user/service.go (hunk 2 above)
Comment (stasBigunenko) [reply service/user/service.go]: remade on two checks

File: service/user/service.go (hunk 1 above)
Comment (stasBigunenko) [reply service/user/service.go]: changed

**Review by DrIhor — APPROVED** [review]

**Review by romanilchyshyn — APPROVED** [review]
LGTM

**Pull request merged**